client/data/themes/with-active-theme.js

```diff
@@ -0,0 +1,18 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useSelector } from 'react-redux';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useActiveThemeQuery } from './use-active-theme-query';
+
+const withActiveTheme = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const siteId = useSelector( getSelectedSiteId );
+		const userLoggedIn = useSelector( isUserLoggedIn );
```

Comment:
Noting that I thought this endpoint might not require a used to be logged in, but I checked and it does--so this is still needed to only query the endpoint when the user is authenticated.